### PR TITLE
Starting Screen

### PR DESCRIPTION
**Pull request by t-scholtz**
Proposed changes
-------------------------------------------
Added a Starting screen which lists all dl stories on local machines and allows player to choose which story to play.

Types of changes
-------------------------------------------
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

Checklist
-------------------------------------------
- [X] My changeset covers only what is described above (no extraneous changes)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged downstream

Further comments
-------------------------------------------
Also changes to setup.py so its prints the name of the DL chosen, as the title of project is printed in the starting screen.
Move all .dl files to Game library.
Added GameAssets folder to store new banner text

### CODE REVIEW

File: GameAssets/bannerIntro.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+*************************************************************************************************************
+
+  .oooooo.   oooo                               o8o                 ooooooooo.   ooooooooo.     .oooooo.    
+ d8P'  `Y8b  `888                               `"'                 `888   `Y88. `888   `Y88.  d8P'  `Y8b   
+888           888   .oooo.    .oooo.o  .oooo.o oooo   .ooooo.        888   .d88'  888   .d88' 888           
+888           888  `P  )88b  d88(  "8 d88(  "8 `888  d88' `"Y8       888ooo88P'   888ooo88P'  888           
+888           888   .oP"888  `"Y88b.  `"Y88b.   888  888             888`88b.     888         888     ooooo 
+`88b    ooo   888  d8(  888  o.  )88b o.  )88b  888  888   .o8       888  `88b.   888         `88.    .88'  
+ `Y8bood8P'  o888o `Y888""8o 8""888P' 8""888P' o888o `Y8bod8P'      o888o  o888o o888o         `Y8bood8P'   
+
```
Comment (emilkovacev) [inline GameAssets/bannerIntro.txt]: This font is awesome! I like the idea of having this in here, is there any way we can store it using pyfiglet instead?

File: GameAssets/bannerIntro.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+`88b    ooo   888  d8(  888  o.  )88b o.  )88b  888  888   .o8       888  `88b.   888         `88.    .88'  
+ `Y8bood8P'  o888o `Y888""8o 8""888P' 8""888P' o888o `Y8bod8P'      o888o  o888o o888o         `Y8bood8P'   
+                                                                                                                                                                                                    
+                            _                                                                                       _                                       
+                           | |__ _  _                               
+                           | '_ \ || |                              
+  __  __      _        ___ |_.__/\_, |     ___                      
+ |  \/  |__ _| |_____ / _ \ _ __ |__/ _ _ / __| ___ _  _ _ _ __ ___ 
+ | |\/| / _` | / / -_) (_) | '_ \/ -_) ' \\__ \/ _ \ || | '_/ _/ -_)
+ |_|  |_\__,_|_\_\___|\___/| .__/\___|_||_|___/\___/\_,_|_| \__\___|
```
Comment (emilkovacev) [inline GameAssets/bannerIntro.txt]: This font is a little difficult to read, and I think I would prefer MakeOpenSource be in text, so that it's as clear as possible. Let me know your ideas though!

File: GameAssets/bannerIntro.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                           |_|                                      
+
```
Comment (emilkovacev) [inline GameAssets/bannerIntro.txt]: We could also add things like the license, and some extra details here!

File: GameAssets/bannerIntro.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+*************************************************************************************************************
```
Comment (emilkovacev) [inline GameAssets/bannerIntro.txt]: The lines are a little off kilt, and can be simplified by the expression `"*" * 100`

File: GameAssets/bannerIntro.txt [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
```
Comment (emilkovacev) [inline GameAssets/bannerIntro.txt]: Files and directories should be written in [snake case](https://en.wikipedia.org/wiki/Snake_case) for consistency!

File: crpg.py [hunk 6]
```diff
@@ -72,7 +72,7 @@ def start(self):
 
 
 def generate(filename):
-    with open(filename, "r") as f:
+    with open("GameLibary/" + filename, "r") as f:
```
Comment (emilkovacev) [inline crpg.py]: Maybe try adding this functionality to a new file that calls the generate function as needed? That way we preserve modularity of the codebase.

File: game-2.py [hunk 7]
```diff
@@ -1,4 +1,10 @@
 from crpg import generate
+from game_loader import selectDL
 
-game = generate("game-2-layout.dl")
-game.start()
+dl = selectDL()
```
Comment (emilkovacev) [inline game-2.py]: It could be really cool if this were run from a CLI, maybe check out the click library!

I'm thinking something like - if we call it on its own, your screen opens, but then we can also pass parameters to it to open specific files? What do you think?

File: game_loader.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    path = "GameLibary"
+    dir_list = os.listdir(path)
+    count = 0
+    options = "(0) - Exit\n"
+    for dl in dir_list:
+        count = count + 1
+        options += "(" +str(count)+ ") - " + dl +"\n"
+    print(options)
+    print("Please choose from the following options")
+    print("_________________________________________")
```
Comment (emilkovacev) [inline game_loader.py]: Can be simplified to "_" * 30

File: setup.py [hunk 9]
```diff
@@ -2,10 +2,10 @@
 from pyfiglet import Figlet
 fonts = ["6x9", "banner", "big", "block", "bubble", "chartr", "chartri", "cour", "digital", "helv", "ivrit", "lean", "mini", "mnemonic", "sbook", "script", "shadow", "slant", "small", "smscript", "smshadow", "smslant", "standard", "term", "times", "xchartr"]
 
-def start():
+def start(gameTitle):
```
Comment (emilkovacev) [inline setup.py]: I love passing the game_title into the setup function!

File: crpg.py [hunk 10]
```diff
@@ -1,7 +1,7 @@
 from setup import start
 import re
 from builtin import Node, Location, Fight, Run, Player
-
+GameTitle = ""
```
Comment (emilkovacev) [inline crpg.py]: maybe GameTitle can be an attribute of a Game object?

File: GameAssets/bannerIntro.txt [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+  .oooooo.   oooo                               o8o                 ooooooooo.   ooooooooo.     .oooooo.    
+ d8P'  `Y8b  `888                               `"'                 `888   `Y88. `888   `Y88.  d8P'  `Y8b
```
Comment (emilkovacev) [inline GameAssets/bannerIntro.txt]: I really like the idea of having a startup screen. I think this will really help the game feel more immersive, thanks for submitting this!

**Review by emilkovacev — CHANGES_REQUESTED** [review]
Thanks for this PR, I really like it! I made a few comments and feature requests, but most of them are small changes. Really great job, hopefully we can merge a lot of your changes soon!

File: GameAssets/bannerIntro.txt (hunk 5 above)
Comment (emilkovacev) [reply GameAssets/bannerIntro.txt]: Additionally, the __pycache__/ file should be deleted. This is on me, I didn't make a .gitignore file for the project

**Comment by t-scholtz** [comment]
I made the requested changes, do I need to make a new pull request, or can you see the changes I made? Sorry, I kinda new to using git. Also I wasn't sure if I should add this to the read.me, but the game can be run from game_2.py, or from game_loader.py with --path allowing a user to enter a path file for a .dl , and --help working, although being very barren at the moment.

**Comment by emilkovacev** [comment]
Awesome, thank you! I can see the changes here, no worries about making a new PR. I'll get to these tomorrow, and get you a review asap! The changes look awesome btw :)

**Comment by emilkovacev** [comment]
Hey there, sorry for the late reply! I looked at your PR, and it is much improved! I have a few suggestions and additions to make just to clean up before we merge, especially since this is such a large PR.

File: crpg.py [hunk 12]
```diff
@@ -2,15 +2,14 @@
...
 # the base crpg game
 
 class Game:
-    def __init__(self, player: Player = Player(), starting_location: Location = None):
+    def __init__(self, gameTitle,  player: Player = Player(), starting_location: Location = None):
         self.current: Node = starting_location
         self.player = player
         self.name: str = None
-
+        self.gameTitle = gameTitle
```
Comment (emilkovacev) [inline crpg.py:12]: I like this change, it makes sense in the context of the game!

File: crpg.py [hunk 13]
```diff
@@ -76,7 +74,10 @@ def generate(filename):
         contents = f.read()
         nodes, connections = contents.split("\n---\n", maxsplit=1)
 
-        game = Game()
+        if(filename.find('/')!=-1):
+            game = Game(filename[filename.find('/')+1:-3])
+        else:
+            game = Game(filename[:-3])
```
Comment (emilkovacev) [inline crpg.py]: Could we pass a filename as an argument to the start method?

File: game-2.py [hunk 14]
```diff
@@ -1,4 +1,10 @@
...
 
-game = generate("game-2-layout.dl")
-game.start()
+dl = start()
+if( str(dl) != "-1"):
+    game = generate(dl)
+    game.start()
+else:
+    print("Exiting")
+    exit()
```
Comment (emilkovacev) [inline game-2.py]: Is this file still needed with the current setup?

File: game_loader.py [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+#Prints Intro Banner - a bit messy
+def bannerPrinter():
+        print("*" * 100)
+        f = Figlet(font = "roman")
+        print(f.renderText("Classic RPG"),end=' ')
+        print("by".center(100))
+        print("MakeOpenSource".center(100))
+        print("University at Buffalo".center(100))
+        print("GPL-3.0 license".center(100))
+        print("*" * 100 + "\n")
```
Comment (emilkovacev) [inline game_loader.py:71]: I see your comment saying "a bit messy", but personally I don't mind this, especially since it's easy to change the content of the banner this way! If you really really wanted to, you could store each of these in variables and only call print once, but it's up to you.

File: game_loader.py [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+from os.path import exists
+
+#Allows adding specifications when starting program; at the moment, only add the abitly to specify exact file path for .dl
+@click.command()
+@click.option("--path", default="game_libary/", help="Provide file path to a .dl file not located under /game_libary")
+def start(path):
+    print(path)
+    #Possible could be more elegant, checks if user specified path; if path exist runs it
+    if(path == "game_libary/"):
+        dl = selectDL(path)
```
Comment (emilkovacev) [inline game_loader.py:19]: can you catch the "FileNotFoundError" and respond with a descriptive message before exiting? (using the function `exit(1)` perhaps)

File: game_loader.py [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+import textwrap
+import click
+from pyfiglet import Figlet
+from crpg import generate
+from functools import reduce
+from os.path import exists
+
+#Allows adding specifications when starting program; at the moment, only add the abitly to specify exact file path for .dl
+@click.command()
+@click.option("--path", default="game_libary/", help="Provide file path to a .dl file not located under /game_libary")
```
Comment (emilkovacev) [inline game_loader.py]: This is perfect! It works exactly as I would want to set the correct path and read from it

File: game_loader.py [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+            if(userIn == "1"):
+                defualtChosen=True
+            else:
+                print("Enter file Path")
+                path = str(input())    
+        if(defualtChosen):
+            path = "game_libary/"
+            dl = selectDL(path)
+        else:
+            dl = path
```
Comment (emilkovacev) [inline game_loader.py]: We could look into how to make this more elegant, like your comment said. 

If it helps, you can add a parameter to make the --path option to be required, that way you can ensure that path always passes something. Then you could use os to check that the directory exists, and then you get a valid directory every time!

File: game_loader.py [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+
+    if( str(dl) != "-1"):
+        game = generate(dl)
+        game.start()
+    else:
+        print("Exiting")
+        exit()
```
Comment (emilkovacev) [inline game_loader.py]: I would specify exit(0)

File: setup.py [hunk 20]
```diff
@@ -2,10 +2,12 @@
...
 fonts = ["6x9", "banner", "big", "block", "bubble", "chartr", "chartri", "cour", "digital", "helv", "ivrit", "lean", "mini", "mnemonic", "sbook", "script", "shadow", "slant", "small", "smscript", "smshadow", "smslant", "standard", "term", "times", "xchartr"]
 
-def start():
+
+
+def start(gameTitle):
     font_idx = random.randint(0, len(fonts) - 1)
     f = Figlet(font=fonts[font_idx])
-    print(f.renderText('Classic RPG'))
+    print(f.renderText(gameTitle))
```
Comment (emilkovacev) [inline setup.py:17]: perfect!

**Review by emilkovacev — CHANGES_REQUESTED** [review]
It's super close to being done! 

There are some super small tweaks I recommended. Your comments were super helpful in knowing what your ideas were too. 

For functions/methods where you weren't certain about the implementation, I tried to give a possible way forward. Let me know what you think!

File: game-2.py (hunk 14 above)
Comment (t-scholtz) [reply game-2.py]: No, the game can be launch by calling the game_loader.py directly. I just wanted to leave this in working condition in case anyone elses stuff needed it.

File: game_loader.py (hunk 16 above)
Comment (t-scholtz) [reply game_loader.py:19]: I set it up so that if it can't find the specified path, it asks the user to enter a new path. Is it still necessary to add a "FileNotFoundError"?

File: game-2.py (hunk 14 above)
Comment (emilkovacev) [reply game-2.py]: In that case, I would revert it to the previous commit, that way the example shows how the API can be used

File: game_loader.py (hunk 16 above)
Comment (emilkovacev) [reply game_loader.py:19]: Could you abstract the entire section to just deal with the path regardless of if it is "game_libary/" with an exception? You could encase it in a while loop so that it always asks only if the exception doesn't go off.

That would help simplify lines 19-42.
